File: BankingApp_WDP.APP/app/src/main/java/com/example/bankingapp/data/model/auth/RegisterRequest.kt
package com.example.bankingapp.data.model.auth

import com.google.gson.annotations.SerializedName

data class RegisterRequest(
    @SerializedName("email")       val email: String,
    @SerializedName("password")    val password: String,
    @SerializedName("firstName")   val firstName: String,
    @SerializedName("lastName")    val lastName: String,
    @SerializedName("phoneNumber") val phoneNumber: String,
    @SerializedName("address")     val address: String,
    @SerializedName("dateOfBirth") val dateOfBirth: String,
    @SerializedName("cnp")         val cnp: String
)

File: BankingApp_WDP.API/DTOs/Auth/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace BankingApp.DTOs.Auth;

public class RegisterRequest
{
    [Required, MaxLength(100)]
    public string FirstName { get; set; } = string.Empty;

    [Required, MaxLength(100)]
    public string LastName { get; set; } = string.Empty;

    [Required, EmailAddress]
    public string Email { get; set; } = string.Empty;

    [Required, MinLength(8)]
    public string Password { get; set; } = string.Empty;

    [Phone]
    public string? PhoneNumber { get; set; }

    [MaxLength(200)]
    public string? Address { get; set; }

    public DateTime? DateOfBirth { get; set; }

    [MaxLength(13)]
    public string? CNP { get; set; }
}
